include/libopencm3/pac55xx/ccs.h

```diff
@@ -0,0 +1,154 @@
+ *  @author Brian Viele <vielster@allocor.tech>
+ *  LGPL License Terms @ref lgpl_license
+ *  @date 1 Dec 2019
+ *
+ *  Definitions in this file come from the PAC55XX Family User Guide Rev 1.21
+ *  by Active-Semi dated August 26, 2019.
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef INCLUDE_LIBOPENCM3_PAC55XX_CCS_H_
+#define INCLUDE_LIBOPENCM3_PAC55XX_CCS_H_
+
+#include <libopencm3/cm3/common.h>
+
+/* clang-format off */
+/* Clock Control Registers */
+/** @defgroup clock_config_regs Clock Config Registers.
+@ingroup system_defines
+@{*/
+#define CCSCTL                        MMIO32(SCC_BASE)
+#define CCSPLLCTL                     MMIO32(SCC_BASE + 0x04)
+#define CCSROSCTRIM                   MMIO32(SCC_BASE + 0x08)
+/**@}*/
+
+/* Port Pin Config Addresses */
+/** @defgroup port_pin_addresses Port Pinmux Register Base.
+@ingroup system_defines
+@{*/
+#define CCS_PORTA                     (SCC_BASE + 0x0C)
+#define CCS_PORTB                     (SCC_BASE + 0x10)
+#define CCS_PORTC                     (SCC_BASE + 0x14)
+#define CCS_PORTD                     (SCC_BASE + 0x18)
+#define CCS_PORTE                     (SCC_BASE + 0x1C)
+#define CCS_PORTF                     (SCC_BASE + 0x20)
+#define CCS_PORTG                     (SCC_BASE + 0x24)
+/**@}*/
+
+/* Port Pin Mux Select Registers */
+/** @defgroup pmux_sel_regs PMUXSEL register mapping.
+@ingroup system_defines
+@{*/
+#define CCS_MUXSELR(base)             MMIO32(base)
+#define CCS_PAMUXSELR                 CCS_MUXSELR(CCS_PORTA)
+#define CCS_PBMUXSELR                 CCS_MUXSELR(CCS_PORTB)
+#define CCS_PCMUXSELR                 CCS_MUXSELR(CCS_PORTC)
+#define CCS_PDMUXSELR                 CCS_MUXSELR(CCS_PORTD)
+#define CCS_PEMUXSELR                 CCS_MUXSELR(CCS_PORTE)
+#define CCS_PFMUXSELR                 CCS_MUXSELR(CCS_PORTF)
+#define CCS_PGMUXSELR                 CCS_MUXSELR(CCS_PORTG)
+#define CCS_MUXSELR_MASK              0x7
+/* clang-format on */
+
+/* Enum type for port function setting for type specificity. */
+typedef enum {
+	CCS_MUXSEL_GPIO = 0,
+	CCS_MUXSEL_AF1  = 1,
+	CCS_MUXSEL_AF2  = 2,
+	CCS_MUXSEL_AF3  = 3,
+	CCS_MUXSEL_AF4  = 4,
+	CCS_MUXSEL_AF5  = 5,
+	CCS_MUXSEL_AF6  = 6,
+	CCS_MUXSEL_AF7  = 7,
+} ccs_muxsel_func_t;
+/**@}*/
+
+/* clang-format off */
```

Comment:
same with everywhere else please, we can't go scattering this all over the place